src/proto/primitive.proto

```diff
@@ -110,32 +95,54 @@ message PowerControl
 
 message Primitive
 {
+    reserved 5;
     oneof primitive
     {
         MovePrimitive move                    = 1;
         StopPrimitive stop                    = 2;
         DirectControlPrimitive direct_control = 3;
     }
-    double cost = 5;
 }
 
 message MovePrimitive
 {
-    MotionControl motion_control          = 1;
-    float final_speed_m_per_s             = 2;
-    Angle final_angle                     = 3;
-    int32 dribbler_speed_rpm              = 4;
-    float max_speed_m_per_s               = 5;
-    AutoChipOrKick auto_chip_or_kick      = 6;
-    float target_spin_rev_per_s           = 7;
-    BallCollisionType ball_collision_type = 8;
-    // Whether the robot should face the direction of intermediate path points
-    // (if there is any) or just the final destination
-    optional bool should_drive_forward = 9;
+    // Fields deprecated after transitioning to trajectory planning
+    reserved 1, 2, 3, 4, 5, 7, 8, 9;
+
+    AutoChipOrKick auto_chip_or_kick        = 6;
+    TrajectoryPathParams2D xy_traj_params   = 10;
+    TrajectoryParamsAngular1D w_traj_params = 11;
+    DribblerMode dribbler_mode              = 12;
 }
 
 message StopPrimitive {}
 
+message TrajectoryPathParams2D
+{
+    // Could remove this and subtract it from destination and
+    // sub_destination. i.e. Trajectories are always from (0,0)
+    Point start_position = 1;
+    // Could create new PointF class (uses floats) or PointS (uses
+    // shorts to represent the milli-meter position, might have to
+    // signed)
+    Point destination                  = 2;
+    Vector initial_velocity            = 3;
+    MaxAllowedSpeedMode max_speed_mode = 4;
+    Point sub_destination              = 5;
+    // If connection_time is non-zero, then the trajectory path is
```

Comment:
Out of curiosity, could we have more that one sub trajectory? What would the problem/advantage be with it?

Reply:
The current implementation of the trajectory planner only supports one sub-destination/trajectory. This is mainly to reduce the complexity of the search for the best trajectory path. Searching for two sub destinations that create the best trajectory path, as opposed to one sub destination will make the search exponentially harder with the current implementation. Additionally, for almost every case, one subdestination is enough to get around obstacles we face for the an initial short term of the trajectory (first couple of seconds), and trajectories generated later into the future will take into account farther obstacles.

But just to future proof it, and to make the code possibly slightly less jank with `connection_time_s = 0` representing no sub-destination, I've made the fields repeated so we can support more complex trajectories with multiple sub-destinations as well.